### ISSUE TITLE
table_wait() sometimes returns before reconfiguration has even begun

### ISSUE THREAD

**Issue by timmaxw** [rethinkdb/rethinkdb#4265]
If a table's config has just been changed and the reconfiguration has not begun, so that it's still available under the old configuration, it should not report `all_replicas_ready`.

I have a script which reproduces this. When it gets checked into GitHub I'll add a link here.

**Comment by larkost** [comment]
For my own reference, the following existing tests fail on this:
```
bonus_tests.reconfigure-under-load-1
continuous.serial
changefeeds.change_master
error_tolerant.failover
error_tolerant.kill-secondary
interface.permanently_remove
bonus_tests.heavy-backfilling
interface.table_acks_durability

```
Note for email readers: this list will increase as I finish my sweep.

**Comment by VeXocide** [comment]
In CR 2935 by @timmaxw.

**Comment by VeXocide** [comment]
Merged into `raft` via commit 19f42a4f5089af3271032acec36d6704c241f637.
